compare_pkglist.py

```diff
@@ -0,0 +1,129 @@
+        self.apiurl = osc.conf.config['apiurl']
+        self.debug = osc.conf.config['debug']
+
+    def get_source_packages(self, project, expand=False):
+        """Return the list of packages in a project."""
+        query = {'expand': 1} if expand else {}
+        root = ET.parse(http_GET(makeurl(self.apiurl,['source', project],
+                                 query=query))).getroot()
+        packages = [i.get('name') for i in root.findall('entry')]
+        
+        return packages
+
+    def check_diff(self, package, old_prj, new_prj):
+        logging.debug('checking %s ...' % package)
+        query = {'cmd': 'diff',
+                 'view': 'xml',
+                 'oproject': old_prj,
+                 'opackage': package}
+        u = makeurl(self.apiurl,['source', new_prj, package], query=query)
+        root = ET.parse(http_POST(u)).getroot()
+        old_srcmd5 = root.findall('old')[0].get('srcmd5')
+        logging.debug('%s old srcmd5 %s in %s' % (package, old_srcmd5, old_prj))
+        new_srcmd5 = root.findall('new')[0].get('srcmd5')
+        logging.debug('%s new srcmd5 %s in %s' % (package, new_srcmd5, new_prj))
+        # Compare srcmd5
+        if old_srcmd5 != new_srcmd5:
+            # check if it has diff element
+            diffs = root.findall('files/file/diff')
+            if diffs:
+                return ET.tostring(root)
+        return False
+
+    def crawl(self):
+        """Main method"""
+        # get souce packages from target
+        print 'Gathering the packages list from %s' % self.old_prj
+        old_packages = self.get_source_packages(self.old_prj)
+        print 'Gathering the packages list from %s' % self.new_prj
+        new_packages = self.get_source_packages(self.new_prj)
+
+        for pkg in old_packages:
+            have_diff = False
+            if pkg not in new_packages:
+                logging.debug('%s is not in %s' % (pkg, self.new_prj))
+            else:
+                have_diff = self.check_diff(pkg, self.old_prj, self.new_prj)
+                if have_diff is not False:
+                    print '%s/%s has different source than %s' % (self.new_prj, pkg, self.old_prj)
+                    if self.verbose:
+                        print have_diff
+
+def main(args):
+    # Configure OSC
+    osc.conf.get_config(override_apiurl=args.apiurl)
+    osc.conf.config['debug'] = args.debug
+
+    uc = CompareList(args.old_prj, args.new_prj, args.verbose)
+    uc.crawl()
+
+if __name__ == '__main__':
+    description = 'Compare packages status between two project'
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument('-A', '--apiurl', metavar='URL', help='API URL')
+    parser.add_argument('-d', '--debug', action='store_true',
+                        help='print info useful for debuging')
+    parser.add_argument('-f', '--old', dest='old_prj', metavar='PROJECT',
+                        help='project where to check (default: %s)' % SLE,
```

Comment:
not: Perhaps include old/new in descriptions for these as well. Is `-f` and `-t` relating to from/to, otherwise perhaps just `-f/-o`?